run/src/types.ts

```diff
@@ -18,36 +18,42 @@ export interface DevToolEnabledSource {
   _isCycleSource: string;
 }
 
-export type Sources = {
-  [name: string]: any;
-};
-
-export type Sinks = {
-  [name: string]: any;
-};
-
-export type SinkProxies<Si extends Sinks> = {[P in keyof Si]: Stream<any>};
+export type SinkProxies<Si> = {[P in keyof Si]: Stream<any>};
 
-export type FantasySinks<Si> = {[S in keyof Si]: FantasyObservable};
-
-export interface Driver<Sink, Source> {
-  (stream: Sink, driverName?: string): Source;
+export interface Driver<Si, So> {
+  (stream: Si, driverName?: string): So;
 }
 
-export type Drivers<So extends Sources, Si extends Sinks> = {
-  [P in keyof (So & Si)]: Driver<Si[P], So[P]>
-};
-
 export type DisposeFunction = () => void;
 
-export interface CycleProgram<So extends Sources, Si extends Sinks> {
-  sources: So;
-  sinks: Si;
+export type Drivers = {
+  [name: string]: (...args: Array<any>) => any;
+};
+export type Main = (...args: Array<any>) => any;
```

Comment:
Shouldnt main just take the sources?

Reply:
This `Main` type for now is quite broad, but it makes sense when used together with all the other stuff. Also I'm explicitly not defining Main as `So => Si` neither a Driver as `Si => So`. Instead, I'm defining Sources as ReturnType<Drivers> and Sinks as ReturnType<Main>